### PR TITLE
Streamline Optimus Documentation

### PR DESCRIPTION
**Pull request by jodihir**
### Purpose
This is PR is for adding public facing documentation of Optimus into this repository, expanding the README file and streamlining Optimus documentation in general. The Optimus report was previously added in another PR.

I updated the table of outputs in the README.txt to match what we have in the List of HCA Pipeline Outputs:
https://docs.google.com/spreadsheets/d/1gKVN7s_66zwQSiZB-r3IbfN18raEArahWAD4wCZnE4c/edit#gid=0

This table and level of detail is currently NOT in the public facing user documentation for Optimus, here: https://prod.data.humancellatlas.org/pipelines/hca-pipelines/data-processing-pipelines/optimus-workflow

### CODE REVIEW

File: pipelines/optimus/PublicDocumentation.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+Overall, the workflow:
+1. corrects cell barcodes and Unique Molecular Identifiers (UMIs)
+2. aligns reads to the genome
+3. generates an expression count matrix in a UMI-aware fashion
+4. detects empty droplets
+5. calculates summary statistics
+6. returns output in BAM and Zarr file formats
+
+Special care is taken to avoid the removal of reads that are not aligned or that do not contain recognizable barcodes. This design (which differs from many pipelines currently available) allows use of the entire dataset by those who may want to use alternative filtering or leverage the data for methodological development associated with the data processing.
```
Comment (ambrosejcarr) [inline pipelines/optimus/PublicDocumentation.md]:
```suggestion
Special care is taken to flag but avoid the removal of reads that are not aligned or that do not contain recognizable barcodes. This design (which differs from many pipelines currently available) allows use of the entire dataset by those who may want to use alternative filtering or leverage the data for methodological development associated with the data processing.
```

File: pipelines/optimus/PublicDocumentation.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+A general overview of the pipeline is shown below, followed by more detailed descriptions of the steps.
+
+![Optimus Pipeline Overview](Optimus_pipeline_overview.png)
+
+## Input Data Preparation 
+
+Each 10X v2 3’ sequencing experiment generates triplets of Fastq files:
+
+1. forward reads (R1), containing the unique molecular identifier and cell barcode sequences
+2. reverse reads (R2), which is the alignable genomic information from the mRNA transcript
```
Comment (ambrosejcarr) [inline pipelines/optimus/PublicDocumentation.md]:
```suggestion
2. reverse reads (R2), which contain the alignable genomic information from the mRNA transcript 
```

File: pipelines/optimus/PublicDocumentation.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+### Cell Barcode Correction
+
+Although the function of the cell barcodes is to identify unique cells, barcode errors can arise during sequencing (such as incorporation of the barcode into contaminating DNA or sequencing and PCR errors), making it difficult to distinguish unique cells from artifactual appearances of the barcode. Barcode errors are evaluated in the [Attach10xBarcodes](https://github.com/HumanCellAtlas/skylab/blob/master/library/tasks/Attach10xBarcodes.wdl) step mentioned above, which compares the sequences against a whitelist of known barcode sequences.
+
+The output file contains the reads with correct barcodes, including barcodes that came within one edit distance ([Levenshtein distance](http://www.levenshtein.net/)) of matching the whitelist of barcode sequences and were corrected by this tool. Correct barcodes are assigned a “CB” tag. Uncorrected barcodes (with more than one error) are preserved and given a “CR” (Cell barcode Raw) tag. Cell barcode quality scores are also preserved in the file under the “CY” tag.
```
Comment (ambrosejcarr) [inline pipelines/optimus/PublicDocumentation.md]:
```suggestion
The output file contains the reads with correct barcodes, including barcodes that came within one edit distance ([Levenshtein distance](http://www.levenshtein.net/)) of matching the whitelist of barcode sequences and were corrected by this tool. Correct barcodes are assigned a “CB” tag. Uncorrectable barcodes (with more than one error) are preserved and given a “CR” (Cell barcode Raw) tag. Cell barcode quality scores are also preserved in the file under the “CY” tag.
```

File: pipelines/optimus/PublicDocumentation.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+## Count Matrix Construction
+
+The pipeline outputs a count matrix that contains, for each cell barcode and for each gene, the number of molecules that were observed. The script that generates this matrix evaluates every read. It discards any read that maps to more than one gene, and counts any remaining reads provided the triplet of cell barcode, molecule barcode, and gene name is unique, indicating the read originates from a single transcript present at the time of lysis of the cell represented by that respective barcode.
```
Comment (ambrosejcarr) [inline pipelines/optimus/PublicDocumentation.md]:
```suggestion
The pipeline outputs a count matrix that contains, for each cell barcode and for each gene, the number of molecules that were observed. The script that generates this matrix evaluates every read. It discards any read that maps to more than one gene, and counts any remaining reads provided the triplet of cell barcode, molecule barcode, and gene name is unique, indicating the read originates from a single transcript present at the time of lysis of the cell.
```

File: pipelines/optimus/PublicDocumentation.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+## Pipeline Output Files
+
+Outputs of the pipeline include:
+1. Raw count matrix
```
Comment (ambrosejcarr) [inline pipelines/optimus/PublicDocumentation.md]:
```suggestion
1. Cell x gene unnormalized count matrix
```

File: pipelines/optimus/PublicDocumentation.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
+# Public Documentation for Optimus
+*The following content about Optimus can also be found on [this page](https://prod.data.humancellatlas.org/pipelines/hca-pipelines/data-processing-pipelines/optimus-workflow) of the DCP Data Portal. This is the master copy.*
+## Introduction to the Optimus Workflow
+
+The long-term goal of the Optimus workflow is to support any 3 prime single cell transcriptomics assay selected by the HCA project. Using the correct modularity, we hope to grow a generic pipeline that has specific modules to address differences in assays, while leveraging common code where steps of the assays are the same. We offer this as a community resource for community development and improvement. The first assay this workflow supports is the [10x v2 (and v3) gene expression assay](https://www.10xgenomics.com/solutions/single-cell/).
```
[the remark this answers is not in the document]
Comment (barkasn) [reply pipelines/optimus/PublicDocumentation.md]: You are right, missed that.

File: pipelines/optimus/PublicDocumentation.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+## Commonalities Among Sequencing Assays
+
+The introduction of droplet-based technologies such as inDrop ([Klein, et al., 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4441768/)) and Drop-seq ([Macosko, et al., 2015](https://www.sciencedirect.com/science/article/pii/S0092867415005498)) moved the throughput of a single-cell RNA sequencing experiment from hundreds to thousands of cells. Technology developed by [10x Genomics](https://www.10xgenomics.com) further increased throughput to hundreds of thousands of cells and has opened up the possibility of creating datasets for millions of cells. Common among many of the single cell transcriptomics high-throughput technologies is the use of:
+
+* microfluidics, which captures individual cells in oil droplets containing barcoded beads and enzymes
+* short read 3’ single-strand DNA sequencing
```
[the remark this answers is not in the document]
Comment (barkasn) [reply pipelines/optimus/PublicDocumentation.md]: That's probably true but we don't support any 3' assay at the moment and he haven't tested for this so let's leave it as it is until then.

File: pipelines/optimus/PublicDocumentation.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+## Quick Start Table
+
+| Pipeline Features | Description | Source |
+|-------------------|---------------------------------------------------------------|-----------------------|
+|Assay Type | 10x Single Cell Expression (v2, v3) |[10x Genomics](https://www.10xgenomics.com)
+| Overall Workflow  |Quality control module and transcriptome quantification module | Code available from [Github](https://github.com/HumanCellAtlas/skylab/blob/master/pipelines/optimus/Optimus.wdl) |
+| Workflow Language |WDL          |[openWDL](https://github.com/openwdl/wdl)|
+| Genomic Reference Sequence|GRCh38 human genome primary sequence|[GENCODE](https://www.gencodegenes.org/human/release_27.html)|
+|Transcriptomic Reference Sequence |V27 GenCode human transcriptome |[GENCODE](https://www.gencodegenes.org/human/release_27.html)|
```
[the remark this answers is not in the document]
Comment (barkasn) [reply pipelines/optimus/PublicDocumentation.md]: STAR does use the gtf file when building the genome, so indirectly we are. But not the reference sequence directly. Do you think we should remove?

[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply pipelines/optimus/PublicDocumentation.md]: Ah ok. Reference _sequence_ to me refers to a _fasta_ file. Can we update this to be a direct link to the gtf file and refer to the reference _annotation_?

File: pipelines/optimus/PublicDocumentation.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+## Quick Start Table
+
+| Pipeline Features | Description | Source |
+|-------------------|---------------------------------------------------------------|-----------------------|
+|Assay Type | 10x Single Cell Expression (v2) |[10x Genomics](https://www.10xgenomics.com)
+| Overall Workflow  |Quality control module and transcriptome quantification module | Code available from [Github](https://github.com/HumanCellAtlas/skylab/blob/master/pipelines/optimus/Optimus.wdl) |
+| Workflow Language |WDL          |[openWDL](https://github.com/openwdl/wdl)|
+| Genomic Reference Sequence|GRCh38 human genome primary sequence|[GENCODE](https://www.gencodegenes.org/human/release_27.html)|
+|Transcriptomic Reference Sequence |V27 GenCode human transcriptome |[GENCODE](https://www.gencodegenes.org/human/release_27.html)|
```
Comment (ambrosejcarr) [inline pipelines/optimus/PublicDocumentation.md]:
```suggestion
|Transcriptomic Reference Annotation |V27 GenCode human transcriptome |[GENCODE](
ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.annotation.gtf.gz)|
```

**Pull request merged**